tocgen.py

```diff
@@ -0,0 +1,20 @@
+from jinja2 import Environment, FileSystemLoader
+import os, re
+
+env = Environment(loader=FileSystemLoader(os.path.dirname('./')))
+template = env.get_template('toc.md.j2')
+
+rootDir = '.'
+
+list_dirs = ""
+for dirName, subdirList, fileList in os.walk(rootDir):
```

Comment:
I'm having trouble following the logic in this block. 

`os.walk()` will recursively walk every sub-directory, and will look for a file named Readme in that directory. But why does it need to walk through the entire directory structure? Each directory in the top level of the repo has a predictable structure - one vignette per folder. (This is reflected in the flat table of contents, too.) It would be simpler to use something like `glob.glob('*')`, which gets a list of files and folders, instead of `os.walk()`, which will do a lot of extra work.

Once you have the list of file and folder names, you can use `os.path.isdir()` on the results of glob to check which ones are directories, and `os.path.isfile()` to verify that those directories contain READMEs.